### PR TITLE
Update Analytics docs

### PR DESCRIPTION
**Pull request by cacheung**
- Update new registration code to index file
- Add Kotlin examples in the index file
- Remove Kotlin Syntax in API doc

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/documentation/adobe-analytics/tabs/index.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
-#### Java
 
 1. Add the [Mobile Core](../mobile-core/index.md) and Analytics extensions to your project using the app's Gradle file.
 
 ```java
 implementation 'com.adobe.marketing.mobile:core:2.x.x'
+implementation 'com.adobe.marketing.mobile:identity:2.x.x'
 implementation 'com.adobe.marketing.mobile:analytics:2.x.x'
 ```
```
Comment (emdobrin) [inline src/pages/documentation/adobe-analytics/tabs/index.md:10]: for consistency can we use the 2.+ notation here and include a warning hint above on how to lock in gradle dependency versions (link to manage-gradle-dependencies)

File: src/pages/documentation/adobe-analytics/tabs/product-variable.md [hunk 2]
```diff
@@ -24,7 +24,24 @@ MobileCore.trackAction("purchase", cdata);
...
+
+```kotlin
+//create a context data dictionary
+val cdata: Map<String, Any?> = mapOf(
+    "&&products" to ";Running Shoes;1;69.95,;Running Socks;10;29.99",
+    "myapp.purchase" to "1",
+    "myapp.purchaseid" to "1234567890"
+)
+
+// send a tracking call - use either a trackAction or TrackState call.
```
Comment (emdobrin) [inline src/pages/documentation/adobe-analytics/tabs/product-variable.md:39]: nit: check the casing - ... trackAction or trackState ...

and thanks for adding the kotlin samples 👍

File: src/pages/documentation/adobe-analytics/tabs/index.md [hunk 3]
```diff
@@ -125,6 +153,22 @@ HashMap cdata = new HashMap<String, Object>();
 // add events
 cdata.put("&&events", "event1:12341234");
 
+// send a tracking call - use either a trackAction or TrackState call.
```
Comment (emdobrin) [inline src/pages/documentation/adobe-analytics/tabs/index.md:156]: trackAction or trackState

**Review by calebk1 — APPROVED** [review]
One small comment. Looks good otherwise!

File: src/pages/documentation/adobe-analytics/tabs/index.md [hunk 4]
```diff
@@ -17,11 +17,22 @@ Using dynamic dependency versions is **not** recommended for production apps. Pl
...
 ```
 
+#### Kotlin
+
+```kotlin
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.Identity
+import com.adobe.marketing.mobile.Analytics
+```
+
```
[the remark this answers is not in the document]
Comment (cacheung) [reply src/pages/documentation/adobe-analytics/tabs/index.md:35]: There is a semi-colon for the sentence in the Java, but not in Kotlin. We have both in Github doc.  @emdobrin, what do you think?

File: src/pages/documentation/adobe-analytics/api-reference.md [hunk 5]
```diff
@@ -133,6 +133,24 @@ iOS
 
 <Tabs query="platform=ios&api=get-visitor-identifier-with-completion-handler"/>
 
+## registerExtension
+
+Registers the Analytics extension with the Mobile Core SDK.
+
+<InlineAlert variant="warning" slots="text"/>
+
+Deprecated as of 2.0.0. Use the [MobileCore.registerExtensions](../mobile-core/api-reference.md#registerextensions) API instead.
```
Comment (calebk1) [inline src/pages/documentation/adobe-analytics/api-reference.md:142]: Oh yeah, one more very minor thing - can we make it "Please use the..." instead? (I like being 🇨🇦 and polite.)

Comment (cacheung) [reply src/pages/documentation/adobe-analytics/api-reference.md]: Sure!

File: src/pages/documentation/adobe-analytics/tabs/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (emdobrin) [reply src/pages/documentation/adobe-analytics/tabs/index.md:35]: let's keep both

File: src/pages/documentation/adobe-analytics/event-reference.md [hunk 6]
```diff
@@ -186,7 +186,7 @@ The following examples show how to use the event data:
 
 ### Analytics request identity
 
-The Analytics request identity event is responsible for fetching the Analytics tracking identifier (AID) and custom visitor identifier (VID) from the Analytics extension and is generated in the Analytics [getTrackingIdentifier](./api-reference.md#gettrackingidentifier) and [getCustomVisitorIdentifier](./api-reference.md#get-the-custom-visitor-identifier) calls.
+The Analytics request identity event is responsible for fetching the Analytics tracking identifier (AID) and custom visitor identifier (VID) from the Analytics extension and is generated in the Analytics [getTrackingIdentifier](./api-reference.md#gettrackingidentifier) and [getCustomVisitorIdentifier](./api-reference.md##setvisitoridentifier) calls.
```
Comment (emdobrin) [inline src/pages/documentation/adobe-analytics/event-reference.md:189]: getvisitoridentifier

File: src/pages/documentation/adobe-analytics/faq.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ The metric `Crashes` is computed based on the Lifecycle start and pause API call
 
 ## How are mobile visits different from launches?
 
-A launch is measured by the SDK when a user opens the app for the first time or returns to the app after having been out of the app for longer than the specified timeout value. The typical timeout is 5 minutes (300 seconds) in the [lifecycleTimeout](../mobile-core/lifecycle/index.md#configuration-keys) configuration setting.
+A launch is measured by the SDK when a user opens the app for the first time or returns to the app after having been out of the app for longer than the specified timeout value. The typical timeout is 5 minutes (300 seconds) in the [lifecycleTimeout](../mobile-core/lifecycle/index.md##device-information) configuration setting.
```
Comment (emdobrin) [inline src/pages/documentation/adobe-analytics/faq.md:25]: double hashtag

File: src/pages/documentation/adobe-analytics/tabs/api-reference.md [hunk 8]
```diff
@@ -384,24 +350,34 @@ Analytics.getVisitorIdentifier { (visitorIdentifier, error) in
...
-<Variant platform="android" api="send-queued-hits" repeat="10"/>
+<Variant platform="android" api="register-extension" repeat="8"/>
 
 #### Java
 
 **Syntax**
 
 ```java
-public static void sendQueuedHits()
+public  static void registerExtension() throws InvalidInitException
```
Comment (emdobrin) [inline src/pages/documentation/adobe-analytics/tabs/api-reference.md:360]: Can you double check on this, Analytics probably does not throw this exception

**Review by emdobrin — COMMENTED** [review]
looks good with few small comments

File: src/pages/documentation/adobe-analytics/tabs/api-reference.md (hunk 8 above)
Comment (cacheung) [reply src/pages/documentation/adobe-analytics/tabs/api-reference.md]: You are right, there is no exception is thrown.

**Review by emdobrin — APPROVED** [review]

**Pull request merged**